cloudferrylib/os/actions/check_networks.py

```diff
@@ -24,65 +24,138 @@
-                # Have the same networks on SRC and DST
-                LOG.debug("SRC network: '%s', DST network: '%s'" %
-                          (network.id, dst_net.id))
-                network.check_network_overlapping(dst_net)
-            else:
-                # Current network does not match with any network on DST
-                # Check Segmentation ID overlapping with DST
-                LOG.debug("Check segmentation ID for SRC network: '%s'",
-                          network.id)
-                network.check_segmentation_id_overlapping(dst_seg_ids)
+        search_opts = kwargs.get('search_opts_tenant', {})
+        search_opts.update({'search_opts': kwargs.get('search_opts', {})})
+
+        src_net_info = NetworkInfo(src_net.read_info(**search_opts))
+        dst_net_info = NetworkInfo(dst_net.read_info())
+        src_compute_info = ComputeInfo(src_compute.read_info(**search_opts))
+
+        # Check subnets and segmentation IDs overlap
+        LOG.debug("Check networks overlapping")
+        nets_overlapped_subnets, nets_overlapped_seg_ids = (
+            src_net_info.get_overlapped_networks(dst_net_info))
+        if nets_overlapped_subnets:
+            overlapped_resources.update(
+                {'networks_with_overlapped_subnets': nets_overlapped_subnets})
+        if nets_overlapped_seg_ids:
+            overlapped_resources.update(
+                {'networks_with_overlapped_segm_ids': nets_overlapped_seg_ids})
 
         # Check floating IPs overlap
         LOG.debug("Check floating IPs for overlapping")
-        for floating_ip in src_info.floating_ips.values():
-            dst_floating_ip = dst_info.floating_ips.get(floating_ip.address)
-            if not dst_floating_ip:
-                LOG.debug("There is no such Floating IP on DST: '%s'. "
-                          "Continue...", floating_ip.address)
-                continue
+        floating_ips = src_net_info.list_overlapped_floating_ips(dst_net_info)
+        if floating_ips:
+            overlapped_resources.update(
+                {'overlapped_floating_ips': floating_ips})
+
+        # Check VMs spawned directly in external network
+        LOG.debug("Check VMs spawned directly in external networks")
+        ext_nets = src_net_info.get_external_networks()
+        vms_list = src_compute_info.list_vms_in_external_network(ext_nets)
+        if vms_list:
+            overlapped_resources.update({'vms_in_external_network': vms_list})
+
+        # Print LOG message with all overlapped stuff and save it to the file
+        if overlapped_resources:
+            LOG.warning('Network overlappings: \n%s', overlapped_resources)
+            utils.write_yaml_file(FILE_NAME, overlapped_resources)
+            raise exception.AbortMigrationError(
+                "There is a number of overlapping Network resources. Please, "
+                "resolve it. Full list of overlapping resources you can find "
+                "in '%s' file." % FILE_NAME)
```

Comment:
ERROR    [/home/zw451w/version1.8/CloudFerry/cloudferrylib/scheduler/scheduler.py : lineno 58]  10-29 02:39:18  BaseTask|CheckNetworks TASK FAILED
Traceback (most recent call last):
  File "/home/zw451w/version1.8/CloudFerry/cloudferrylib/scheduler/scheduler.py", line 71, in process_chain
    self.run_task(task)
  File "/home/zw451w/version1.8/CloudFerry/cloudferrylib/scheduler/scheduler.py", line 63, in run_task
    self.map_func_task[task](task)
  File "/home/zw451w/version1.8/CloudFerry/cloudferrylib/scheduler/scheduler.py", line 94, in task_run
    task(namespace=self.namespace)
  File "/home/zw451w/version1.8/CloudFerry/cloudferrylib/scheduler/task.py", line 97, in __call__
    result = self.run(**namespace.vars)
  File "/home/zw451w/version1.8/CloudFerry/cloudferrylib/os/actions/check_networks.py", line 92, in run
    "in '%s' file." % FILE_NAME)
AbortMigrationError: There is a number of overlapping Network resources. Please, resolve it. Full list of overlapping resources you can find in 'check_networks_result.yaml' file.
INFO     [/home/zw451w/version1.8/CloudFerry/cloudferrylib/scheduler/scheduler.py : lineno 76]  10-29 02:39:18  Failed processing CHAIN PREPARATION